wherehows-web/app/components/dataset-aclaccess.js

```diff
@@ -0,0 +1,138 @@
+import Component from '@ember/component';
+import { computed, getProperties, setProperties, get, set } from '@ember/object';
+import { requestAclAccess } from 'wherehows-web/utils/api/datasets/acl-access';
+import { accessState, pickList } from 'wherehows-web/constants/dataset-aclaccess';
+
+export default Component.extend({
+  /**
+   * Define property to bind textarea
+   * @type {string}
+   */
+  requestReason: '',
+
+  /**
+   * Define the property for the initial page  
+   * @type {object} 
+   */
+  accessInfo: null,
+
+  /**
+   * Define the property for reloading the page when a user applies for permission.
+   * @type {object} 
+   */
+  accessResponse: null,
+
+  /**
+   * Define the property to identify user 
+   * @type {string} 
+   */
+  currentUser: null,
+
+  /**
+   * Define the property for loading spinner component
+   * @type {boolean} 
+   */
+  isLoadSpinner: false,
+
+  /**
+   * Define the property for deciding component state. The component has 5 states.
+   * @type {string}
+   */
+  pageState: computed('accessInfo', 'accessResponse', function() {
+    const { accessInfo, accessResponse } = getProperties(this, ['accessInfo', 'accessResponse']);
+
+    if (accessInfo) {
```

Comment:
I agree with @cptran777 , this `pageState` computed property should be refactored, rather than having this branching logic. It will be a hassle to maintain.